src/RetryProxy.php

```diff
@@ -69,4 +84,9 @@ public function call(callable $action, array $arguments = []): ?mixed
 
         throw new RetryException('Action call is failed.');
     }
+
+    public function setLogger(Logger $logger): void
```

Comment:
add this method to the interface? otherwise the code will have dependency on RetryProxy instead of RetryProxyInterface
and if adding to interface, maybe we can have it as optional ctor arg?

Reply:
If we put it in the constructor, I don't think we really need the setter so I'll get rid of it.

Reply:
possibly